biscuit-parser/src/builder.rs

```diff
@@ -157,10 +157,15 @@ impl ToTokens for Scope {
             Scope::Authority => quote! { ::biscuit_auth::builder::Scope::Authority},
             Scope::Previous => quote! { ::biscuit_auth::builder::Scope::Previous},
             Scope::PublicKey(pk) => {
-                let bytes = pk.iter();
-                quote! { ::biscuit_auth::builder::Scope::PublicKey(
-                  ::biscuit_auth::PublicKey::from_bytes(&[#(#bytes),*]).unwrap()
-                )}
+                let bytes = pk.key.iter();
+                match pk.algorithm {
+                    Algorithm::Ed25519 => quote! { ::biscuit_auth::builder::Scope::PublicKey(
+                        ::biscuit_auth::PublicKey::from_bytes(&[#(#bytes),*], ::biscuit_auth::builder::Algorithm::Ed25519).unwrap()
```

Comment:
the coverage issue could be fixed by adding a relevant test in the macros module